src/agent/onefuzz/src/monitor.rs

```diff
@@ -94,9 +94,17 @@ impl DirectoryMonitor {
                         .ok_or_else(|| format_err!("missing path for file create event"))?
                         .clone();
 
-                    if !self.report_directories && fs::metadata(&path).await?.is_dir() {
-                        // Ignore
-                    } else {
+                    if self.report_directories {
+                        return Ok(Some(path));
+                    }
+
+                    let is_dir = fs::metadata(&path)
+                        .await
+                        .ok()
+                        .map(|f| f.is_dir())
+                        .unwrap_or_default();
```

Comment:
is this unwrap correct? if the path is missing we will unwrap the failed result to `false`, so the path will still be returned from the function

Reply:
good catch, i changed the logic to check if it is a file instead

Reply:
Additionally, I think we probably want the following behavior:
1. Try to get the `Metadata`.
2. If `Ok(..)`, then report (conditionally if `report_directories` is `true`).
3. If `Err(err)` and `err.kind() == io::ErrorKind::NotFound`, we can assume we've lost a race with deletion so, ignore the file. Otherwise, something genuinely weird happened at the IO layer, so propagate the error (callers can always ignore it if desired).

Reply:
@ranweiler I opted to emit a warning instead of stopping the monitoring in case the error is intermittent